### PR TITLE
[polaris.shopify.com] Add components lifecycle

### PR DESCRIPTION
**Pull request by sarahill**
### WHY are these changes introduced?

We have component statuses within individual component pages but no place where we document what each status means.

### What is this PR doing?

This PR adds a component lifecycle page with definitions, requirements, and expectations for each stage.

[Google doc content](https://docs.google.com/document/d/1v_dlfosdBxMYlfnJ1HOff5lS8Rj4VVXkib1W3-99daU/edit#)

### CODE REVIEW

**Comment by sarahill** [comment]
One weird thing is the section titles. I had wanted them to be the same for each section: "Requirements" and "What to expect" but there's a bug with the TOC that causes weird selected state issues when there are duplicate section titles. That's why they're longish and repetitive. Open to suggestions. cc: @jjgali if you have any thoughts

image

**Comment by sarahill** [comment]
The page lives under "Polaris 101" right now. I was debating between "Polaris 101" and "Contributing". I landed on 101 because the content also applies to how our team works and feels necessary to understand when using components in general. That being said I'm happy to move it if others have strong opinions that it should live elsewhere.

image

File: polaris.shopify.com/content/getting-started/components-lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  - alpha
+  - beta
+  - stable
+  - deprecated
+order: 1
+---
+
+### Feedback and suggestions
+
+Have a question, feedback or suggestion? You can [create an issue](https://github.com/Shopify/polaris/issues/new) during any stage of a component’s lifecycle.
```
Comment (selenehinkley) [inline polaris.shopify.com/content/getting-started/components-lifecycle.md]:
```suggestion
Have feedback, a question, or suggestion? [Create an issue](https://github.com/Shopify/polaris/issues/new) during any stage of a component’s lifecycle.
```

File: polaris.shopify.com/content/getting-started/components-lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+---
+
+## Alpha
+
+The component is a work in progress and ready for exploratory usage, with breaking changes expected. During this stage, we’re being critical about understanding the purpose of the component and what problem it’s trying to solve. If the purpose is unclear, we may decide not to move forward with the component and it could be removed without warning.
+
+### Requirements for alpha
+
+- Has a clear purpose and rationale for why it belongs in Polaris.
```
Comment (selenehinkley) [inline polaris.shopify.com/content/getting-started/components-lifecycle.md]:
```suggestion
- Has a clear purpose and rationale for why it belongs in Polaris
```

Comment (selenehinkley) [reply polaris.shopify.com/content/getting-started/components-lifecycle.md]: If the entire list is made up of single sentences, we don't punctuate. All the periods from this list can be removed :)

File: polaris.shopify.com/content/getting-started/components-lifecycle.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+- Uses Polaris [tokens](https://polaris.shopify.com/tokens) with no references to any hard-coded static values.
+- Designed with responsiveness in mind and can adapt across [breakpoints](https://polaris.shopify.com/tokens/breakpoints).
+- Meets known [accessibility](https://polaris.shopify.com/foundations/accessibility) requirements like: contrast, interactive states, touch target size, etc.
+- In progress documentation exists that includes the list of props with some examples.
+- Documentation includes examples in Storybook.
+- The component has 100% test coverage.
+
+### What to expect during alpha
+
+- Usage is being evaluated, and may not progress to Beta. Evaluation includes any necessary research and data that supports why the component belongs in Polaris.
```
Comment (selenehinkley) [inline polaris.shopify.com/content/getting-started/components-lifecycle.md]:
```suggestion
- Usage is being evaluated, and may not progress to beta. Evaluation includes any necessary research and data that supports why the component belongs in Polaris.
```

File: polaris.shopify.com/content/getting-started/components-lifecycle.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+The component moves to beta once there’s a clear understanding of the problem and we’re confident in the solution. The focus of this stage is all about testing in enough environments to make sure the component is bug free.
+
+Ready for wider adoption, encouraged for most cases. Breaking changes are possible in minor version updates.
+
+### Requirements for beta
+
+- Used multiple times in production (Shopify admin).
+- Meets all [accessibility](https://polaris.shopify.com/foundations/accessibility) requirements for our use cases.
+- Usage guidelines and documentation covers common use cases.
+- Has been reviewed and tested in production by designers and developers.
```
Comment (selenehinkley) [inline polaris.shopify.com/content/getting-started/components-lifecycle.md]:
```suggestion
- Reviewed and tested in production by designers and developers.
```

File: polaris.shopify.com/content/getting-started/components-lifecycle.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+- Figma components available in the Polaris Components file.
+
+### What to expect during beta
+
+- Work is being done to move to Stable.
+- Manual and automated migration guidance is being worked on.
+
+## Stable
+
+The component is bug free and works in most if not all environments. It’s ready for full adoption, with long-term support expected.
```
Comment (selenehinkley) [inline polaris.shopify.com/content/getting-started/components-lifecycle.md]:
```suggestion
The component is bug free and works in most, if not all, environments. It’s ready for full adoption, with long-term support expected.
```

**Review by tracytees — APPROVED** [review]

**Review by selenehinkley — COMMENTED** [review]
Did a light copy edit. Didn't focus on the structure or accuracy of the content. Overall, use oxford commas and don't punctuate lists if they're all single sentences. If a list item has more than one sentence, then the entire list "chunk" is punctuated :)

**Comment by jjgali** [comment]
@sarahill This looks great!

Re: the bug with section titles, it's incredibly annoying, but I think you've got a good work around here. The other option would be to have shorter, repeatable titles, and ignore the bug until there's a fix (that's my current plan of action for a different page I'm working on). 

I'm not sure what the timeline on fixing that [bug](https://github.com/Shopify/polaris/issues/6974) is (seems like it might be a little complex), so I'll defer to your best judgement. I don't think these titles are too long winded if you decide to keep them. 

Re: living in the Polaris 101 section, I think that makes sense for now. I'd almost expect this to live within the Components section, but I don't think we have the page structure to support that right now.

**Review by jjgali — APPROVED** [review]

**Pull request merged**